docs/how-tos/how-to-use-searchkick-for-search-autocomplete.mdx

```diff
@@ -0,0 +1,285 @@
+        end
+
+        def search_data
+          { name: name }
+        end
+
+        ::Spree::Taxon.prepend self
+      end
+    end
+  end
+end
+```
+
+We can now run a reindex for `Spree::Taxon` and verify it's
+working:
+
+```ruby title="bin/rails c"
+Spree::Taxon.reindex
+Spree::Taxon.search("ruby").map(&:name)
+# => ["Ruby"]
+```
+
+## Enhancing search autocomplete
+
+[Solidus Starter Frontend][ssf] comes with a simple search autocomplete out of
+the box. In this section of the guide we will change its behavior to use the
+newly added advanced search functionalities.
+
+:::info
+
+The search autocomplete feature has been added to the Starter Frontend
+since Solidus 3.4.0. If you don't have it, please refer to
+[Autocomplete main search with products and taxons][autocomplete-pr] to
+backport that feature in your storefront.
+
+:::
+
+To let the autocomplete controller use Searchkick, we need to change the
+code of the `AutocompleteResultsController` provided like this:
+
+```diff title="app/controllers/autocomplete_results_controller.rb"
+
+  def autocomplete_products
+    if params[:keywords].present?
+-      searcher = build_searcher(params.merge(per_page: 5))
+-      searcher.retrieve_products
++      Spree::Product.search(params[:keywords], fields: [{name: :word_start}], limit: 5)
+    else
+      Spree::Product.none
+    end
+  end
+
+  def autocomplete_taxons
+    if params[:keywords].present?
+-       Spree::Taxon
+-         .where(Spree::Taxon.arel_table[:name].matches("%#{params[:keywords]}%"))
+-         .limit(5)
++       Spree::Taxon.search(params[:keywords], fields: [{name: :word_start}], limit: 5)
+    else
+      Spree::Taxon.none
+    end
+  end
+```
+
+With just these small changes, the autcomplete will finally use Searchkick
+under the hood.
+
+
+## Next steps
+
+We just stratched the surface of what's possible with Searchkick, and
+it's left to each store to implement the configuration that better suits
+its needs. Still, there are some big things worth considering and in
+this section we'll see the most relevant for a typical Solidus store.
+
+:::info
+
+[Searchkick README][searchkick-github] is really well done and contains
+more details about the information reported in this guide. Please, always
+refer to that documentation to get extended and updated information.
+
+:::
+
+
+### Reindex Strategies
+
+By default, everytime you change a record, if Searchkick is enabled
+on that model, a reindex will be triggered for that record only.
+While this is very useful, it can hurt stores perfomances.
+
+To mitigate this problem, it is suggested to use the Asynchronous
+Reindex Strategy, which will push the update to a background job.
+
+To enabale this strategy, you can edit the directive that enabled
+Searchkick on the models, like this:
+
+```diff title="app/overrides/my_store/spree/product/add_searchkick.rb"
+-   base.searchkick word_start: [:name]
++   base.searchkick word_start: [:name], callbacks: :async
```

Comment:
Do you think we should point out that the worker will need further setup?

Reply:
Good catch, adding:

```
:::info

Please refer to Rails' [Active Job][active-job] guide to setup your
system for background jobs support.

:::
```

Do you think it's enough?

Reply:
That's great 👍